### DISCUSSION TITLE
[RFC] Form.Item help supports render type

### DISCUSSION THREAD

**Discussion by yunho1017** [ant-design/ant-design#48001] [RFCs]
## Motivation

I want to use help props as guide message for the input. (always visible)
so i use help props, I cant't see the validate error when after validation.
i want to use help and validate together.

So I added a render type that passes errors to params.

### As-is

code 

```
    <ProFormText
        label={"name"}
        name={"name"}
        help={"You can also input emojis and special character"}
        fieldProps={{ style: { width: "100%" } }}
        rules={[Validations.required()]}
    />
```

before validate
<img width="409" alt="스크린샷 2024-03-26 오후 2 48 52" src="https://github.com/ant-design/ant-design/assets/22070720/b1c8679e-799b-48e3-9976-408c0836720d">

after validate
<img width="378" alt="스크린샷 2024-03-26 오후 2 49 02" src="https://github.com/ant-design/ant-design/assets/22070720/6df3f442-f43c-4944-8bcf-ab1829dd68cc">

After validation, help is displayed instead of the required error message.

### To-be

code 

```
    <ProFormText
        label={"name"}
        name={"name"}
        help={({errors}) => errors.length ? errors[0] : "You can also input emojis and special character"}
        fieldProps={{ style: { width: "100%" } }}
        rules={[Validations.required()]}
    />
```

before validate
<img width="409" alt="스크린샷 2024-03-26 오후 2 48 52" src="https://github.com/ant-design/ant-design/assets/22070720/71958608-f7db-4e3f-8a06-4e19fc8f5eb6">

after validate
<img width="432" alt="스크린샷 2024-03-26 오후 2 48 28" src="https://github.com/ant-design/ant-design/assets/22070720/0e7a88a5-afd5-4338-ae1a-ef68af5125d8">

After validation, I want to show an error message.

## API

```
help?:
    | React.ReactNode
    | ((params: { errors: React.ReactNode[]; warnings: React.ReactNode[] }) => React.ReactNode);
```

## Basic Example
```
<Form>
  <Form.Item label="hello" help={() => 'bamboo'}>
    <Input />
  </Form.Item>
</Form>
```

### PR
https://github.com/ant-design/ant-design/pull/47753

### Issue
https://github.com/ant-design/ant-design/issues/47994

**Comment by afc163** [comment]
There should be more description about Motivation and Alternatives

**Comment by yunho1017** [comment]
@afc163 
Sorry for the late reply.
i have  added content.

If it's not enough, please tell me again

**Comment by zombieJ** [comment]
Seems it could be enhanced on the ProComponents. The antd side provided the Form.ErrorList component. So it can be realized with HOC to pass the validation info to the Form.ErrorList

**Reply by yunho1017** [reply]
@zombieJ 
Hello !
You said use Form.ErrorList,
I checked docs that Form.ErrorList requires errors to be passed as props.
Is there a way to get the errors of the current field from outside?

**Comment by yunho1017** [comment]
Oh, I see. Then, can I contribute at pro-components?

**Reply by poyiding** [reply]
yes you can

**Reply by yunho1017** [reply]
All right ! I will close that PR and create a new PR in proComponent!
